lib/rules/decorator-position.js

```diff
@@ -1,16 +1,91 @@
 'use strict';
 
-const { CLIEngine: v7AndEarlier, ESLint: v8Plus } = require('eslint');
+/**
+ * @import { TSESLint, TSESTree } from '@typescript-eslint/utils'
+ * @import { Options as PrettierOptions } from 'prettier'
+ *
+ * @typedef {{
+ *   ESLint: typeof import('eslint').ESLint;
+ *   CLIEngine?: typeof import('eslint').ESLint;
+ * }} EslintPackageUniversal
+ *
+ * @typedef { string | [string, { withArgs: boolean }] } DecoratorConfigRaw
+ *
+ * @typedef {{
+ *   printWidth?: number;
+ *   properties?: 'prefer-inline' | 'above';
+ *   methods?: 'prefer-inline' | 'above';
+ *   overrides?: {
+ *     above?: Array<DecoratorConfigRaw>;
+ *     'prefer-inline'?: Array<DecoratorConfigRaw>;
+ *   };
+ * }} RuleOptionsRaw
+ *
+ * @typedef {{
+ *   printWidth: number;
+ *   properties: 'prefer-inline' | 'above';
+ *   methods: 'prefer-inline' | 'above';
+ *   overrides: {
+ *     above: Array<DecoratorConfigRaw>;
+ *     'prefer-inline': Array<DecoratorConfigRaw>;
+ *   };
+ * }} RuleOptionsNormalized
+ *
+ * @typedef { 'same-line' | 'different-line' } DecoratorConfigIntent
+ *
+ * @typedef {{
+ *   withArgs?: boolean;
+ *   intent?: DecoratorConfigIntent | undefined;
+ * }} DecoratorConfigOptionsNormalized
+ *
+ * @typedef { [string, DecoratorConfigOptionsNormalized ] } DecoratorConfigNormalized
+ *
+ * @typedef { Pick<RuleOptionsNormalized, 'printWidth'> & { overrides: Pick<RuleOptionsNormalized['overrides'], 'prefer-inline'> } } OptionsPlaceDecoratorsBesideProperty
+ *
+ * @typedef { Pick<RuleOptionsNormalized, 'printWidth'> & { overrides: Pick<RuleOptionsNormalized['overrides'], 'above'> } } OptionsPlaceDecoratorsAboveProperty
+ *
+ * @typedef {{
+ *   needsTransform?: boolean;
+ *   name?: string;
+ *   decorator?: any;
+ *   arity?: any;
+ *   arityMatches?: any;
+ *   currentPositionMatchesIntent?: any;
+ *   inlineLength?: any;
+ *   ifInlineWouldViolatePrettier?: any;
+ * }} DecoratorInfo
+ *
+ * @typedef { 'expectedInline' | 'expectedAbove' } MessageId
+ *
+ * @typedef { TSESLint.RuleModule<MessageId, [RuleOptionsRaw]> } RuleModule
+ *
+ * @typedef { Omit<RuleModule, 'defaultOptions' | 'meta'> & { meta: Omit<RuleModule['meta'], 'schema'> } } RuleModuleOmited
+ *
+ * @typedef { Parameters<RuleModule['create']>[0] } RuleModuleCreateContext
+ *
+ * @typedef { TSESTree.PropertyDefinition | TSESTree.MethodDefinition } DefinitionNode
+ *
+ * @typedef { TSESTree.PropertyNameComputed | TSESTree.ClassPropertyNameNonComputed } DecoratorKey
+ */
+
+const { CLIEngine: v7AndEarlier, ESLint: v8Plus } = /** @type {EslintPackageUniversal} */ (
+  require('eslint')
+);
 
 const ESLint = v7AndEarlier || v8Plus;
 const isEslint7 = Boolean(v7AndEarlier);
 // const isEslint8 = !isEslint7;
 
 const cli = new ESLint();
 
+/**
+ * @param { string } filePath
+ * @returns { { rules?: Record<string, Record<string | number, any>> } }
```

Comment:
The helper does not appear to be required for strict typing, as it is targeted at older versions of eslint.